File: python/lib/dynamodb.py
import boto3

# https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/dynamodb.html

class CustomDynamoDB:
  def __init__(self):
    pass

  def resource():
    return boto3.resource('dynamodb', endpoint_url='http://localhost:8000')

  def client():
    return boto3.client('dynamodb', endpoint_url='http://localhost:8000')

File: ruby/lib/dynamodb.rb
# frozen_string_literal: true

require 'aws-sdk-dynamodb'

# https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/DynamoDB/Client.html
# https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/DynamoDB/Resource.html

# Create client or resource of dynamodb
class CustomDynamoDB
  @@region = 'ap-northeast-1'
  @@endpoint = 'http://localhost:8000'

  def self.resource
    Aws::DynamoDB::Resource.new(region: @@region, endpoint: @@endpoint)
  end

  def self.client
    Aws::DynamoDB::Client.new(region: @@region, endpoint: @@endpoint)
  end
end
